interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a JavaScript library used for creating user interfaces. It is made up of components that are analogous functions and allow for reusability throughout the application. It is used to build a wide variety of projects ranging from web pages to apps, and allows for scalability, which is incredibly important as applications become more and more popular and gain more users. 
 
-  Researched answer:
+  Researched answer: React is known for its scalability, dynamicness, and reusability. One of the most awesome features is the virtal DOM, which allows changes to occur without updating entire webpages. The virtual DOM allows components to be updated without needing to refresh the entire page, which keeps the page from slowing down as well.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The built-in render() method is the required lifecycle method in a React class component. Render() automatically runs and updates components without refreshing the entire page in a browser.
 
-  Researched answer:
+  Researched answer: We cannot change state or cause side effects in the render method. The render methods allow us to read props and state as well as return JSX code to the root component of the app. 
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a mix of html and Javascript languages used to code in react to create dynamic applications for user interfaces. HTML uses class and JSX uses classNames.
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScript XML and is an extension of Javascript that allows us to intertwine HTML with logic by using react. It is faster than regular Javascript. In HTML, multiple elements can be returned, but in JSX only one element can be returned, so we create nests and use fragments to return multiple elements.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager created by Facebppk to manage dependency files. The files that are modifies are package.json and yarn.lock. Yarn is a way to tie everything together.
 
-  Researched answer:
+  Researched answer: Yarn stands for Yet Another Resource Negotiator. It's a package manager that allows developers to use and share code around the world. It doubles as a project manager. Package.json is a manifest that describes the package. Yarn.lock is the main source of information about the current versions of dependencies in a project.
```

Comment:
Overall, when running yarn, you are updating the node modules in your application.